### PR TITLE
Add support for tag method

### PR DESCRIPTION
**Pull request by tduffield**
Add support for tag method for registries using the v2 API. 

* Add an exception to throw when using an unsupported version of the Registry API. 
* Add tests for adding and removing tags. 

This PR also fixes a bug in the `split_auth_header` library that would fail if `www-authenticate` contained a comma in one of the values, which it commonly will. 

Supersedes #25 as it includes tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,8 +232,11 @@ The following exceptions are thrown:
 
 * `RegistryAuthenticationException`: username and password are invalid
 * `RegistryAuthorizationException`: registry does not support pushing the layers or uploading the manifest using the given credentials, probably because the repository is private and the credentials provided do not have access
+* `RegisteryVersionException`: You're not running a registry that supports the v2 API.
```
Comment (deitch) [inline README.md]: Please change this to "registry does not support the v2 API". The person running the lib here might not be running the registry; could be against a corporate or cloud registry.

File: docker_registry2.gemspec [hunk 2]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(/^(test|spec|features)\//)
   spec.require_paths = ['lib']
 
-  spec.add_development_dependency 'bundler', '~> 1.6'
+  spec.add_development_dependency 'bundler'
```
Comment (deitch) [inline docker_registry2.gemspec]: Why remove the constraint? Did it get in the way?

Comment (tduffield) [reply docker_registry2.gemspec]: It did. Unless I wanted to create an entirely new ruby gemset/installation that was just for Bundler 1. Any modern ruby installation is going to default to Bundler 2. Looking at the gemspec, I didn't see a reason to restrict it.

Comment (deitch) [reply docker_registry2.gemspec]: Fair enough. Truth is, I cannot remember why I restricted it back then. :-)

**Comment by deitch** [comment]
Thanks @tduffield . It should get pushed out by CI/CD shortly. If it isn't, let me know, and I will push manually.

**Pull request merged**